pandas/tools/pivot.py

```diff
@@ -136,8 +136,14 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
 
     table = agged
     if table.index.nlevels > 1:
-        to_unstack = [agged.index.names[i] or i
-                      for i in range(len(index), len(keys))]
+        index_names = agged.index.names[:len(index)]
+        to_unstack = []
+        for i in range(len(index), len(keys)):
+            name = agged.index.names[i]
```

Comment:
Sorry for being MIA on this PR. 

Unfortunately, things are getting busy on my end so I don't think I will have time implement a private argument to `stack()` (and probably `unstack()` as well) to indicate passing level or label indexers. I could add it on the original issue as the ideal solution.

I might be able to play around with this loop to make it cleaner.